### ISSUE TITLE
The example scrollers exhibit strange behaviour (don't work well)

### ISSUE THREAD

**Issue by akarelas** [jankuca/ng-scroller#6]
If you go to http://ngscroller.herokuapp.com/examples/twitter.html with either Firefox (on Linux) or Chrome (on Linux), then this happens:

Scroll some amount downwards (as much as you want)

Scroll back upwards - the scrollbar never reaches the start, while the scrolling div doesn't let me scroll upwards more than a small amount.

Now try to scroll downwards: it will only scroll a tiny bit, won't let you scroll any further, and you're trapped (in both directions)

**Comment by jankuca** [comment]
Yeah, I know about this bug. I hope to get to fixing that next week.

**Comment by akarelas** [comment]
Hi, how are you doing?

I was just wondering whether there's been any progress with this bug

Thanks

**Comment by akarelas** [comment]
Hi any news?

**Comment by akarelas** [comment]
Hi, Are there plans to fix this bug, or have you abandoned the project? I'm writing some software that might need ng-scroller, and was wondering whether I should rely on it or not.

**Comment by jankuca** [comment]
I moved to implementing Object.observe into AngularJS for now but I will definitely try to fix those bugs as I will need this component for my apps too.

I'm not sure when will that be exactly though, probably next month. I was debugging this for a few days and couldn't figure out what the problem is so I'm not really eager to do it asap.

**Comment by unbalanced** [comment]
Your blog post really gave high hopes for a top notch solution and the de-facto of cross platform high pref scrolling for angular.
Hope you could get back to it. and keep on building great stuff! thanks.

**Comment by apuchkov** [comment]
I also got excited by reading a blog post and seeing more than 100 stars for this repo. Unfortunately scrolling up doesn't work well in any of demos.

**Comment by kristianmandrup** [comment]
I think most of the problems could easily be resolved by using Functional Reactive Programming (FRB) using Bacon.js
https://github.com/baconjs/bacon.js

Complex stream handling from multiple event streams are very difficult to handle using conventional programming paradigms and often leads to buggy behaviour. Unfortunately I'm not really a Bacon expert myself, but this is clearly the way forward for the new (more) event driven apps and components that we see.

**Comment by axelitus** [comment]
I was excited while reading your post on https://medium.com/p/463bc649c7bd but as stated above when trying the demos they don't work very well. The post though explains very well the concepts behind and the idea of this scroller is awesome... it just needs time and effort to mature I hope...

**Comment by kristianmandrup** [comment]
Try out ngInfiniteScroll at http://binarymuse.github.io/ngInfiniteScroll/ and https://github.com/sroze/ngInfiniteScroll

Basic demo http://binarymuse.github.io/ngInfiniteScroll/demo_basic.html

I'm trying my best to make a demo app, but still newbie with Angular

https://github.com/kristianmandrup/ngInfinite-demo-app

Feel free to assist and get it working ;)
